lexos/static/js/scripts_kmeans.js

```diff
@@ -1,172 +1,139 @@
-$(function () {
-  // Show the silhouette score results based whether the results are shown
-  if ($('#kmeansresults').length) {
-    $('#silhouetteResults').show()
+/**
+ * The function to run the error modal.
+ * @param {string} htmlMsg: the message to display.
+ * @returns {void}.
+ */
+function runModal (htmlMsg) {
+  $('#error-modal-message').html(htmlMsg)
+  $('#error-modal').modal()
+}
+
+/**
+ * At least two documents are required to run the K-Means clustering.
+ * @returns {string | null}: the errors that is checked by JS, if no error the result will be null.
+ */
+function submissionError () {
+  if ($('#num_active_files').val() < 3) {
+    return 'You must have at least 2 active documents to proceed!'
   } else {
-    $('#silhouetteResults').hide()
+    return null
   }
-  function doAjax(action) {
-    $.ajax({
-      'complete': function (response) {
-        rand = Math.round(Math.random() * 100) + 1
-        $('#plotlyFrame').attr('src', $('#plotlyFrame').attr('src') + '?' + rand)
-        $('#plotlyFrame').hide()
-        // Handle labels separately (eventual solution to the Unicode problem)
-        $.each(response['responseJSON']['labels'], function (index, label) {
-          lb.push(label)
-        })
-        lb = JSON.stringify(lb)
-        var bdiv = response['responseJSON']['bdiv']
-        bdiv = bdiv.replace('"ticktext":', '"tickangle":90,"ticktext":')
-        var re = /"ticktext": \[.+?\]/
-        bdiv = bdiv.replace(re, '"ticktext": ' + lb)
-        $('.PCAImage').html(bdiv)
+}
+
+/**
+ * The function to convert the form into json.
+ * @returns {{string: string}}: the form converted to json.
+ */
+function jsonifyForm () {
+  const form = {}
+  $.each($('form').serializeArray(), function (i, field) {
+    form[field.name] = field.value || ''
+  })
+  return form
+}
+
+/**
+ * Send the ajax request.
+ * @param {string} url: the url to post.
+ * @param {{string: string}} form: the form data packed into an object.
+ * @returns {jQuery.Ajax}: an jQuery Ajax object.
+ */
+function sendAjaxRequest (url, form) {
+  return $.ajax({
+    type: 'POST',
+    url: url,
+    contentType: 'application/json; charset=utf-8',
+    data: JSON.stringify(form)
+  })
+}
+
+/**
+ * Return the desired data table configuration.
+ * @returns {{paging: boolean, scrollY: number, scrollCollapse: boolean, dom: string, buttons: string[], columnDefs: *[]}} data table configuration.
+ */
+function getDataTableConfiguration () {
+  const hideColumns =
+    $('#vizMethod').val() === '3DScatter' ? [2, 3, 4] : [2, 3]
+
+  return {
```

Comment:
I don't think this is a good idea. imaging a typed system, when you type `{pag<tab>}`, you will not get any intellisense, since the code analyzer don't know what object you are creating.

But if you type `table.DataTable({pag<tab>})`, you will get intellisense here, because the code analyzer knows what type `.DataTable()` method takes.

Therefore I recommend you to change to the following code:
```js
function makeDataTable(table) {
  const stuff = stuff
  table.DataTable({paging: false, other stuff..})
}
```